lib/metasploit/framework/login_scanner/x3.rb

```diff
@@ -0,0 +1,115 @@
+# -*- coding: binary -*-
+
+# require 'msf/core'
+# require 'msf/core/exploit/tcp'
+require 'metasploit/framework/login_scanner/base'
+require 'metasploit/framework/login_scanner/rex_socket'
+require 'metasploit/framework/tcp/client'
+
+module Metasploit
+  module Framework
+    module LoginScanner
+      class X3
+        include Metasploit::Framework::LoginScanner::Base
+        include Metasploit::Framework::LoginScanner::RexSocket
+        include Metasploit::Framework::Tcp::Client
+
+        DEFAULT_PORT = 50000
+        REALM_KEY = nil
+
+        def encrypt_pass(inp)
+          # check if it's already encrypted
+          if inp.include?('CRYPT:')
+            return inp
+          end
+
+          num2 = inp.length
+          num = 17
+          ret = ''
+          charset_0 = 'cromanwqxfzpgedkvstjhyilu'.split('')
+          xyz = 'zxWyZxzvwYzxZXxxZWWyWxYXz'.split('')
+          charset_1 = 'cf2tln3yuVkDr7oPaQ8bsSd4x'.split('')
+
+          (0..num2 - 1).each do |i|
+            num5 = inp[i].ord
+            num7 = num5.to_f / num.to_f
+            num10 = (num5 % num)
+            num11 = xyz[i].ord
+            num12 = num11 - num7
+            if num12.to_i != num12
+              num12 += 1
+            end
+            ret += num12.to_i.chr
+            ret += charset_0[num10].ord.chr
+            off = charset_0.find_index(ret.split('').to_a[-1])
+            if off & 1 == 0
+              ret += charset_1[off].ord.chr
```

Comment:
When concatenating strings, the `<<` operator is preferred vs the `+=` operator.

Reply:
Made this change in [dadc59e](https://github.com/deadjakk/metasploit-framework/commit/dadc59ebbe66e376f78da31c4236cee78841f841)